app/models/game_night.rb

```diff
@@ -1,4 +1,10 @@
 class GameNight < ActiveRecord::Base
   validates :time, presence: true
   validates :location_name, presence: true
+
+  belongs_to :group
+  belongs_to :organizer, class_name: :User, foreign_key: :organizer_id
+
+  has_many :user_game_night_attendees
+  has_many :users, through: :user_game_night_attendees
```

Comment:
Hey @npauzenga, making `Membership` or `Attendance` more generic would make it easier if we need to reutilize that same functionality. If we don't have a case where we would need to reutilize it then keeping them specific works great :smile:.

If we wanted to generalize them, we could use polymorphic associations :ghost: :snake:. These shapeshifting creatures are documented [here](http://guides.rubyonrails.org/association_basics.html#polymorphic-associations). ActiveRecord can handle everything for us but on a high level, all it's doing is adding two attributes to the table: an **id** and a **type**. We tell AR the name of our polymorphic friend, for example `attendable` and it uses that to add the column `attendable_id`, then it automatically figures out the type of object based on its class name and uses that for the `attendable_type` column. So when we query for `user.attendances` it's checking something like:

```ruby
# This is probably not exactly how AR's implementation looks
Attendance.where(attendable_id: user.id, attendable_type: user.class.name.downcase)
```